org.eclipse.wildwebdeveloper/src/org/eclipse/wildwebdeveloper/debug/AbstractRunHTMLDebugTab.java

```diff
@@ -156,13 +210,18 @@ public void initializeFrom(ILaunchConfiguration configuration) {
 	@Override
 	public void performApply(ILaunchConfigurationWorkingCopy configuration) {
 		String programPath = this.programPathText.getText();
-		configuration.setAttribute(AbstractHTMLDebugDelegate.PROGRAM, programPath);
+		if (programPathText.isEnabled()) {
+			configuration.setAttribute(AbstractHTMLDebugDelegate.PROGRAM, programPath);
+		} else if (urlText.isEnabled()) {
+			configuration.setAttribute(ChromeRunDAPDebugDelegate.URL, urlText.getText());
+		}
+
 		configuration.setAttribute(AbstractHTMLDebugDelegate.ARGUMENTS, this.argumentsText.getText());
 		String workingDirectory = this.workingDirectoryText.getText();
 		configuration.setAttribute(AbstractHTMLDebugDelegate.CWD, workingDirectory);
 		configuration.setAttribute(DebugPlugin.ATTR_WORKING_DIRECTORY, workingDirectory);
 		configuration.setMappedResources(ResourcesPlugin.getWorkspace().getRoot().findFilesForLocationURI(new File(programPath).toURI()));
-	
+		configuration.setAttribute(AbstractHTMLDebugDelegate.URL_DEBUG, urlRadio.getSelection());
```

Comment:
Same as above; I don't think we need to store that as the configuration shouldn't keep URL in case File is used, nor File when URL is used.